### PR TITLE
Implement timing distribution in Swift

### PR DESCRIPTION
**Pull request by badboy**

### CODE REVIEW

File: glean-core/ios/Glean/Glean.swift [hunk 1]
```diff
@@ -8,6 +8,9 @@ private typealias GleanBaseline = GleanMetrics.GleanBaseline
 private typealias GleanInternalMetrics = GleanMetrics.GleanInternalMetrics
 private typealias Pings = GleanMetrics.Pings
 
+/// Public exported type identifying individual timers for `TimingDistributionMetricType`
+public typealias GleanTimerId = UInt64
```
[the remark this answers is not in the document]
Comment (mdboom) [reply glean-core/ios/Glean/Glean.swift]: On glean-ac (at least), this was used in both Timespan and TimingDistribution in the timing manager.  That's all moved to Rust now, but I still think is a higher level concept than just a single metric...

[the remark this answers is not in the document]
Comment (badboy) [reply glean-core/ios/Glean/Glean.swift]: yeah, this just replicates what the android side is also doing at the moment

File: glean-core/ios/Glean/Metrics/DistributionData.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        } catch {
+            return nil
+        }
+    }
+
+    /// The total count of accumulated values.
+    ///
+    /// This is calculated from all recorded values.
+    var count: UInt64 {
+        values.reduce(0) { (acc, tuple: (_key: UInt64, value: UInt64)) in
```
Comment (travis79) [inline glean-core/ios/Glean/Metrics/DistributionData.swift]: I ❤️ `reduce()` and the other nifty array transformation functions in Swift

File: glean-core/ios/Glean/Metrics/TimingDistributionMetric.swift [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    /// Stop tracking time for the provided metric and associated timer id. Add a
+    /// count to the corresponding bucket in the timing distribution.
+    /// This will record an error if no `start()` was called.
+    ///
+    /// - parameters:
+    ///     * timerId: The `GleanTimerId` to associate with this timing.
+    ///                This allows for concurrent timing of events associated with different ids
+    ///                to the same timespan metric.
+    public func stopAndAccumulate(_ timerId: GleanTimerId?) {
+        // `start` might return null.
```
Comment (travis79) [inline glean-core/ios/Glean/Metrics/TimingDistributionMetric.swift]: Swift nit: `null` -> `nil`

File: glean-core/ios/Glean/Metrics/TimingDistributionMetric.swift [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        // Accepting that means users of this API don't need to do a null check.
```
Comment (travis79) [inline glean-core/ios/Glean/Metrics/TimingDistributionMetric.swift]: Swift nit: `null` -> `nil`

File: glean-core/ios/Glean/Metrics/TimingDistributionMetric.swift [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        return glean_timing_distribution_test_has_value(Glean.shared.handle, self.handle, pingName).toBool()
+    }
+
+    /// Returns the stored value for testing purposes only. This function will attempt to await the
+    /// last task (if any) writing to the the metric's storage engine before returning a value.
+    ///
+    /// Throws a "Missing value" exception if no value is stored
+    ///
+    /// -parameters:
+    ///   * pingName: represents the name of the ping to retrieve the metric for.
```
Comment (travis79) [inline glean-core/ios/Glean/Metrics/TimingDistributionMetric.swift]: super nit: indentation of list item doesn't match other function parameter docs, looks like it's missing a space?

**Pull request merged**